common/model/db/tables/record/table.js

```diff
@@ -25,3 +31,29 @@ export default class TableRecord extends TableSurvey {
 }
 
 TableRecord.columnSet = columnSet
+
+/**
+ * Transforms the result row of a query on the record table into a Record object.
+ *
+ * @param {!number} surveyId - The ID of the survey.
+ * @param {boolean} [includeValidationFields=true] - Whether to include all the Validation fields.
+ * @returns {Record} - The record object.
+ */
+TableRecord.dbTransformCallback = (surveyId, includeValidationFields = true) => (record) => {
+  const validation = Record.getValidation(record)
+  // camelize all but 'validation'
+  return R.pipe(
+    R.dissoc(Validation.keys.validation),
+    camelize,
+    R.assoc('surveyId', surveyId),
+    R.assoc(
+      Validation.keys.validation,
+      includeValidationFields
+        ? validation
+        : {
+            ...Validation.newInstance(Validation.isValid(validation)),
+            [Validation.keys.counts]: Validation.getCounts(validation),
+          }
+    )
+  )(record)
```

Comment:
this can just be:
```
const {validation, ...rest} = record
return {
surveyId,
...camelize(rest),
[Validation.keys.validation]:  includeValidationFields ? validation : {
              ...Validation.newInstance(Validation.isValid(validation)),
              [Validation.keys.counts]: Validation.getCounts(validation),
            }
}
```

Reply:
ok